Makefile

```diff
@@ -81,12 +83,12 @@ UARCH_TO_SHARE= uarch-ram.bin
 MONGOOSE_VERSION=7.12
 
 # Build settings
-DEPDIR := third-party
-SRCDIR := $(abspath src)
-DOWNLOADDIR := $(DEPDIR)/downloads
-DEPDIRS := third-party/mongoose-$(MONGOOSE_VERSION)
-SUBCLEAN := $(addsuffix .clean,$(SRCDIR) uarch third-party/riscv-arch-tests)
-COREPROTO := lib/grpc-interfaces/core.proto
+DEPDIR = third-party
```

Comment:
Why change to "="? Does the value of any variable get reassigned between evaluations?

Reply:
The use `:=` forces evaluation of commands once and immediately, while `=` only when the variable is instantiated. This is a problem when doing cross compilation and some command that is not available in the system could be skipped.

A good example is this line:
```
GRPC_PROTOBUF_LIB=$(shell pkg-config --libs grpc++ protobuf)
````
In case we used `:=` it would require invoking `pkg-config --libs grpc++ protobuf` even when not compiling grpc stuff, and throw compile errors.

I made a refactor to replace almost all `:=` with `=` to make the Makefile more cross compilation friendly.